### ISSUE TITLE
Make download preparation asynchronous

### ISSUE THREAD

**Issue by phraenquex** [m2ms/fragalysis-frontend#1258]
Need entry created in drop-down as soon as "Prepare" is clicked.  @boriskovar-m2ms describes complexities...

Copied from #1256:
Outstanding functionality from #1075 is to add download immediately to the download combo box after the preparation process started and handle all possible fault states accordingly. Optimistic updates are required and if possible (i.e. feature is moved from experimental to production) use useOptimistic hook.

**Comment by phraenquex** [comment]
@boriskovar-m2ms is this now fixed?  Is it still worth fixing?

**Comment by boriskovar-m2ms** [comment]
I don't think this is worth it. It's about having download that is only preparing already in the drop down. Not really even sure why that would be useful but I'm sure that it will be complicated because the dialog is implemented to use directly opposite approach how to handle things (pessimistic vs optimistic)

**Comment by phraenquex** [comment]
Okay, let's drop it then, thanks
